src/codegen/wasm.zig

```diff
@@ -55,18 +55,60 @@ pub fn genCode(buf: *ArrayList(u8), decl: *Decl) !void {
     // Reserve space to write the size after generating the code
     try buf.resize(5);
 
-    // Write the size of the locals vec
-    // TODO: implement locals
-    try leb.writeULEB128(writer, @as(u32, 0));
+    // Calculate the locals of the body
+    // We must calculate the occurance of each value type and emit the count of it
+    const tv = decl.typed_value.most_recent.typed_value;
+    const mod_fn = tv.val.castTag(.function).?.data;
+
+    var locals: [4]u8 = [_]u8{0} ** 4;
+    for (mod_fn.body.instructions) |inst| {
+        if (inst.tag != .alloc) continue;
+
+        const alloc = inst.castTag(.alloc).?;
+        const elem_type = alloc.base.ty.elemType();
+
+        switch (genValtype(elem_type)) {
+            0x7F => locals[0] += 1,
+            0x7E => locals[1] += 1,
+            0x7D => locals[2] += 1,
+            0x7C => locals[3] += 1,
+            else => unreachable,
+        }
```

Comment:
This could just be
```suggestion
        // valtypes are 0x7C, 0x7D, 0x7E, or 0x7F
        locals[genValtype(elem_type) - 0x7C] += 1;
```
If the code below is updated in a similar fashion.